merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -17,12 +19,14 @@ export class Constraint {
     })
   }
 
-  public static ForEachActivity(activityType: Gen.ActivityTypeName, alias: string, constraint: Constraint): Constraint {
+  public static ForEachActivity<A extends Gen.ActivityTypeName>(activityType: A, expression: (instance: Gen.ActivityInstance<A>) => Constraint): Constraint {
```

Comment:
What does this new interface achieve over the old one? Why are local ActivityInstances useful?

Reply:
No management of aliases, which were very difficult to type check. Plus I think its more intuitive when making reference to activity properties

Reply:
The previous approaches had the user refer to activity instance aliases by a string name. For example, to forbid usage of an activity, you'd do this:

```
export default(): Constraint => Constraint.ForEachActivity(
    "MyActivity",
    "activity instance alias",
    Windows.During("activity instance alias").not()
)
```

If you happened to mis-type the alias (i.e. `Windows.During("some other alias")`), there would be no immediate error and it would fail in the backend. After all of the changes in this PR, you would now write it as this:

```
export default(): Constraint => Constraint.ForEachActivity(
    ActivityType.MyActivity,
    instance => instance.window().not()
```

This hands the problem of validating activity alias names to the typescript compiler. It also makes it easier to know the schema of the activity instance parameters. E.g. if the activity has a `someNumber` parameter, we can now access it as `instance.parameters.someNumber`, which will carry enough info about the parameter's type to do compile-time validation.